components/dashboard/src/repositories/detail/variables/ConfigurationVariableList.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useState } from "react";
+import { ConfigurationSettingsField } from "../ConfigurationSettingsField";
+import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { Button } from "@podkit/buttons/Button";
+import { ModifyVariableModal } from "./ConfigurationModifyVariableModal";
+import { Heading3, Subheading } from "@podkit/typography/Headings";
+import { Table, TableBody, TableHead, TableHeader, TableRow } from "@podkit/tables/Table";
+import { useListConfigurationVariables } from "../../../data/configurations/configuration-queries";
+import { LoadingState } from "@podkit/loading/LoadingState";
+import { ConfigurationVariableItem } from "./ConfigurationVariableItem";
+
+type Props = {
+    configuration: Configuration;
+};
+export const ConfigurationVariableList = ({ configuration }: Props) => {
+    const { data, isLoading } = useListConfigurationVariables(configuration.id);
+    const [showAddVariableModal, setShowAddVariableModal] = useState<boolean>(false);
+
+    if (isLoading || !data) {
+        return <LoadingState />;
+    }
+
+    return (
+        <ConfigurationSettingsField>
+            {showAddVariableModal && (
+                <ModifyVariableModal
+                    configurationId={configuration.id}
+                    onClose={() => {
+                        setShowAddVariableModal(false);
+                    }}
+                />
+            )}
+            <div className="mb-2 flex">
+                <div className="flex-grow">
+                    <Heading3>Environment Variables</Heading3>
+                    <Subheading>Manage repository-specific environment variables.</Subheading>
+                </div>
+            </div>
+            {data.length === 0 ? (
+                <div className="bg-gray-100 dark:bg-gray-800 rounded-xl w-full p-6 flex flex-col justify-center space-y-3">
+                    <span className="font-bold text-base">No environment variables are set</span>
+                    {/* todo: change to podkit color abstractions */}
+                    <span className="text-sm text-gray-500 dark:text-gray-400">
+                        All configuration-specific environment variables will be visible in prebuilds and optionally in
+                        workspaces for this repository.
+                    </span>
+                </div>
+            ) : (
+                <Table>
+                    <TableHeader>
+                        <TableRow>
+                            <TableHead>Name</TableHead>
+                            <TableHead>Visibility</TableHead>
```

Comment:
I think adding a fixed width to this column and the one below (that holds the dropdown menu button) will help some layout shift as someone adds more variables. Then the name column will grow to whatever is left over. A strategy I try to do with tables is to have all but 1 column a fixed width when possible.

Reply:
Making the width fixed here sounds reasonable. I'm not sure whether adding it to the one below will change anything, since it's always the same width?

Reply:
I see, because it by default reserves more place than it needs. Nice